### ISSUE TITLE
False positive on `Layout/CommentIndentation`

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubocop-hq/rubocop#6450]
When using the `rails` style of `Layout/IndentationWidth`, I get a `Layout/CommentIndentation` offense for the following code:

```ruby
module Helpers

  protected

    # One method.
    #
    def one_method
      does_stuff
    end

    # Another method.
    #
    # def another_method
    #   does_something_else
    # end

  private

    # Yet another method.
    #
    def yet_another_method
      does_yet_another_thing
    end
end
```

--------

## Expected behavior

No `Layout/CommentIndentation` offenses.

## Actual behavior

A `Layout/CommentIndentation` offense.

## Steps to reproduce the problem

Copy the previous code to a `test.rb` file and the following config

```yaml
---

AllCops:
  DisabledByDefault: true

Layout/CommentIndentation:
  Enabled: true

Layout/IndentationConsistency:
  Enabled: true
  EnforcedStyle: rails
``` 

to a `.rubocop_test.yml` file, and run `bundle exec rubocop test.rb -c .rubocop_test.yml`.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.5.3 x86_64-linux)
```
